commands/deploy_command.go

```diff
@@ -221,24 +225,95 @@ func (c *DeployCommand) Execute(args []string) ExecutionStatus {
 	mtaElementsCalculator := mtaElementsToAddCalculator{shouldAddAllModules: false, shouldAddAllResources: false}
 	mtaElementsCalculator.calculateElementsToDeploy(optionValues)
 
-	mtaArchive, err := getMtaArchive(parser.Args(), mtaElementsCalculator)
+	rawMtaArchive, err := getMtaArchive(parser.Args(), mtaElementsCalculator)
 	if err != nil {
 		ui.Failed("Error retrieving MTA: %s", err.Error())
 		return Failure
 	}
 
+	isUrl := false
+	mtaArchive := ""
+	switch castedMtaArchive := rawMtaArchive.(type) {
```

Comment:
Is it possible to extract this switch case in other function to decrease the size of the current one?